### DISCUSSION TITLE
riscv arch 架构编译

### DISCUSSION THREAD

**Discussion by youngday** [xmake-io/xmake#3029] [Ideas]
### 你在什么场景下需要该功能？

在riscv64架构的C906核心的哪吒板D1开发，debian,ubuntu近期都推出了riscv的版本，openeuler也推出了预览版，
安装openeuler后，目前只有系统安装包，是否有支持riscv64的xmake-repo计划呢？感谢！

### 描述可能的解决方案

riscv64的xmake-repo编译包

**Comment by waruqi** [comment]
切到交叉编译平台，走交叉编译，原本就支持

```lua
xrepo install -p cross --sdk=/xxx/linux-riscv64-sdk zlib
```

```lua
xmake -f p cross  --sdk=/xxx/linux-riscv64-sdk 
xmake
```

具体哪些包支持交叉编译，可以看下 https://xrepo.xmake.io/#/packages/cross

还不支持的包，也可以提 pr 过来，改进包配置，放开 cross 平台支持。

**Reply by youngday** [reply]
感谢，后面遇到非cross包，再向您提交了，计划切换conan到xmake.
